### ISSUE TITLE
node-spiffe can't be used inside Express?

### ISSUE THREAD

**Issue by phelicksg** [depot/node-spiffe#55]
Hi,

apologies if this is a newbie mistake, as I'm not very fluent in Express/Node yet.
There is one example which works great, when used standalone.
However, that example relies on using the "import" statement, which is not allowed when used in an Express app. Instead, one would have to rewrite the example to use the "require" statement.

I am banging my head against the wall with the line:

for await (const message of rpc.responses) {

I've changed the first lines to:

```
var spiffe = require ('spiffe');
// Connect to a specific endpoint
const client = spiffe.createClient('unix:///tmp/spire-agent/public/api.sock')

// Read x509 credentials from Workload API
const rpc = client.fetchX509SVID()

for await (const message of rpc.responses) {
```
[...]

Which renders the error:

```
usr/src/app/bad.js:14
for await (const message of rpc.responses) {
    ^^^^^

SyntaxError: Unexpected reserved word
    at wrapSafe (node:internal/modules/cjs/loader:1469:18)
    at Module._compile (node:internal/modules/cjs/loader:1491:20)
    at Module._extensions..js (node:internal/modules/cjs/loader:1691:10)
    at Module.load (node:internal/modules/cjs/loader:1317:32)
    at Module._load (node:internal/modules/cjs/loader:1127:12)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:217:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:166:5)
    at node:internal/main/run_main_module:30:49

Node.js v22.9.0
```

Obviously I am doing something wrong, but this is not trivial. Do you think you could provide a short example that uses require instead of import to use the SPIFFE library? That would be super appreciated. Thanks so much in advance.

**Comment by jacobwgillespie** [comment]
Hey, no worries!

So the issue here is actually the `await` in `for await` - that's [a JavaScript language feature](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of) that loops over an "async iterable", which is basically an iterable (object that produces multiple values) that produces each next value as a promise that needs to be `await`'d. Kinda like a stream of values. In the example, `for await (const message of rpc.responses)` basically says "for each response from the API".

Since it's using `await` though, you will likely need to run it inside an async function. There is something called "top level await" support that lets you use `await` outside of async functions, but it's experimental in the latest versions of Node.

That could look something like this:

```ts
const spiffe = require('spiffe');

// Connect to a specific endpoint
const client = spiffe.createClient('unix:///tmp/spire-agent/public/api.sock')

async function main() {
  // Read x509 credentials from Workload API
  const rpc = client.fetchX509SVID()

  for await (const message of rpc.responses) {
    ...
  }
}

main().catch(err => {
  console.error(err)
})
```

Basically you will want to use the `for await` inside of an `async` function, either a new one you make or an existing one in your codebase.

`import` vs `require` is [a whole separate topic ](https://blog.logrocket.com/commonjs-vs-es-modules-node-js/), but this library should support both ways the same.

**Comment by phelicksg** [comment]
Hi Jacob,

thanks so much for that! This is super helpful. Indeed, I am new to Node although I've been scripting for decades using other languages. My initial problem seemed to be that I had to use require because my mini-Express server was CommonJS rather than a model. Independently, I rewrote my server to be a module that could use import, and in that case, of course it worked right out of the box.

If I may make a suggestion, it would be to update the documentation and give an example that would work in the CommonJS example. If you want, I can take a stab at this, and let you choose whether you want to add it or not.

Also, with regards to the doc, there are some very helpful utility functions that you added. I found them after trying to decode a certificate chain coming from SPIFFE itself for a few hours, and was elated to find that you already created them. Specifically, I'm talking about parseCertificateBundle, which BTW you wouldn't just used for the svid.bundle but also for the svid.x509Svid (in case you're getting a chain back for the x509Svid, which you would if they're signed with an intermediate cert chained to a root cert in your PKI).

So my offer stands, I can take a stab at completing the docs and getting that over to you for consideration? May take a bit of time, but I will if you'd like.

Thanks again!
